kotlin/lib/src/main/kotlin/goodmetrics/Client.kt

```diff
@@ -139,11 +140,11 @@ fun Any.asDimension(): Dimension = when (this) {
     }
 }
 
-fun Set<Map.Entry<String, Any>>.initializeDatum(timestampNanos: Long, name: String): Datum = datum {
+fun MetricPosition.initializeDatum(timestampNanos: Long, name: String): Datum = datum {
     unixNanos = timestampNanos
     metric = name
-    for ((metric, position) in this@initializeDatum) {
-        dimensions[metric] = position.asDimension()
+    for (position in this@initializeDatum) {
+        dimensions[metric] = position.value.asDimension()
```

Comment:
this needs to be `dimensions[position.name]`. see https://github.com/WarriorOfWire/goodmetrics_kotlin/issues/7 🤦

Reply:
Thank you for the catch!